core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -861,6 +858,18 @@ private static void deletePhysicalPartition(List<PartitionSpec> partitionSpecs,
     }
   }
 
+  /**
+   * This method deletes the directories recursively if there are no files under that folder
```

Comment:
This comment does not describe the function precisely.
As a matter of fact, you are trying to delete the parent directory too, but the comment here does not point that out.

Reply:
besides I think there is no need to add another funtion here.
just keep the old code and only add deletion for path's parent.

Reply:
here i have added function because, i need to traverse recursively and check and delete the parent folder if there are not files present under that folder.

Reply:
i have changed comment